src/UI/Component/Icon/Icon.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Icon;
+
+/**
+ * This describes how a icon could be modified during construction of UI.
+ */
+interface Icon extends \ILIAS\UI\Component\Component {
+
+	// sizes of icons
+	const SMALL = 'small';
+	const MEDIUM = 'medium';
+	const LARGE = 'large';
+
+	/**
+	 * Get the CSS-class of the icon.
+	 *
+	 * @return string
+	 */
+	public function getCSSClass();
```

Comment:
Please rename to `getName`. We try not to talk about HTML/CSS in the API of the UI framework. Also, in this case, as far as I understand, this also is not a simple CSS-class that multiple but distinct icons could have. It really is the name of the icons, since this class is used to establish the uniqueness of the icon.